server/service/frontend.go

```diff
@@ -101,44 +107,63 @@ func ServeEndUserEnrollOTA(
 			return
 		}
 
-		fs := newBinaryFileSystem("/frontend")
-		file, err := fs.Open("templates/enroll-ota.html")
-		if err != nil {
-			herr(w, "load enroll ota template: "+err.Error())
+		errorMsg := r.URL.Query().Get("error")
+		if errorMsg != "" {
+			if err := renderEnrollPage(w, appCfg, urlPrefix, "", errorMsg); err != nil {
+				herr(w, err.Error())
+			}
 			return
 		}
 
-		data, err := io.ReadAll(file)
-		if err != nil {
-			herr(w, "read bindata file: "+err.Error())
+		enrollSecret := r.URL.Query().Get("enroll_secret")
+		if enrollSecret == "" {
+			// TODO(mna): how should we pass this formatted error with heading? Needs to be special-cased on the frontend I think...
+			// https://www.figma.com/design/fw7XXg2QzBOa7YJ9r2Cchp/-29222-IdP-authentication-before-BYOD-iOS--iPadOS--and-Android-enrollment?node-id=5319-3340&t=mn243Pavsxvkd4QC-0
+			if err := renderEnrollPage(w, appCfg, urlPrefix, "", "This URL is invalid. : Enroll secret is invalid. Please contact your IT admin."); err != nil {
+				herr(w, err.Error())
+			}
 			return
 		}
 
-		t, err := template.New("enroll-ota").Parse(string(data))
+		authRequired, err := requiresEnrollOTAAuthentication(r.Context(), ds,
```

Comment:
@MagnusHJensen  This is the reusable bit you can call to do the checks to see if an IdP UUID must be provided along with the enroll secret.